Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/RelationshipsAuthorizationStrategyBase.cs

```diff
@@ -116,20 +116,26 @@ string GetNonRoleNamedProperty(string propertyName)
             {
                 //Look for the corresponding non - role named property on the model using its DefiningConcreteProperty
 
-                var schema = authorizationContext.Type.GetCustomAttribute<SchemaAttribute>(false)?.Schema;
-                var expectedEntityFullName = new FullName(schema, authorizationContext.Type.Name);
-
-                if (!_domainModel.EntityByFullName.TryGetValue(expectedEntityFullName, out var entity))
+                return _nonRoleNamedByRoleNamedPropertyName.GetOrAdd(propertyName, _ =>
```

Comment:
Consider using the parameter to the delegate rather than the outer `propertyName` variable to avoid unnecessary use of a closure.

Reply:
Fixed it